src/vulkan-cts-runner.cc

```diff
@@ -406,12 +405,18 @@ bool process_block(Context &ctx, unsigned thread_id) {
     } else if (string_matches(line, "Test case '")) {
       auto len = strlen(line) - 3;
       auto name = std::string(line + 11, line + len);
-      auto it = state.indices.find(name);
-      if (it == state.indices.end()) {
-        abort();
+
+      for (;;) {
+        if (state.indices.empty()) {
+          std::cerr << "Unexpected test " << name << "\n";
+          abort();
+        } else {
+          state.test_idx = state.indices.back();
```

Comment:
I think this does not work in all cases.

dEQP-VK does (or did at some point) tests in alphabetical order even if the caselist is not in alphabetical order. AFAIU this code assumes deqp runs tests in order.

(Unless I missed some sorting)

Reply:
Ugh, yeah.  I'd discovered that at one point and failed to write a test for it.